datanode/api/trading_data_v2.go

```diff
@@ -138,9 +138,13 @@ func (t *tradingDataServiceV2) ObserveAccounts(req *v2.ObserveAccountsRequest,
 		t.log.Debug("Accounts subscriber - new rpc stream", logging.Uint64("ref", ref))
 	}
 
-	return observe(ctx, t.log, "Accounts", accountsChan, ref, func(account entities.AccountBalance) error {
+	return observeBatch(ctx, t.log, "Accounts", accountsChan, ref, func(accounts []entities.AccountBalance) error {
+		protoAccounts := make([]*vega.Account, len(accounts))
+		for i := 0; i < len(accounts); i++ {
+			protoAccounts[i] = accounts[i].ToProto()
+		}
```

Comment:
Why not just use a range expression here + append here? Not that it changes much things ultimately though.

Reply:
doesn't that make an extra copy of the account object? I mean I doubt it makes much difference but I was trying to be a good little efficient go boy :)

Reply:
doesn't that make an extra copy of the account object? I mean I doubt it makes much difference but I was trying to be a good little efficient go boy :)

Reply:
nope normally the generated assembly would be pretty much the same, if not the same after optimisations.

Reply:
Was curious so I had a play at https://godbolt.org/ with this little program

```
package main

type Thingy struct {
    X int
    Y string
}

func main() {
    thingies := make([]Thingy, 100)
    wotsits := make([]Thingy, len(thingies))

    // for i := 0; i<len(thingies); i++ {
    //    wotsits[i] = thingies[i]
    // }

    for _, thing := range(thingies){
        wotsits = append(wotsits, thing)
    }
    _=wotsits    
}
```

It's been a while since I had to figure out what some generated assembler is doing but I think when using the second on it is making a temporary copy of each thing as it iterates through the list - there's this bit of assembler that is in the range() version but not in the simple for loop:

```
       MOVQ    BX, main.wotsits.len+64(SP)
        MOVQ    DX, main..autotmp_17+88(SP)
        MOVQ    CX, main..autotmp_18+104(SP)
        MOVQ    R9, main..autotmp_19+80(SP)
        MOVQ    R8, main..autotmp_20+96(SP)
        MOVQ    R10, main..autotmp_21+72(SP)
```

In any case it's a bit academic :)

Reply:
Thingy is also not a pointer here, while in your code they are pointers, but yea Thingy is tiny here anyway.

Reply:
To be fair, I've approved I don't mind either

Reply:
Thanks 

FWIW the thing I am iterating over is a a `[]entities.AccountBalance`, not `[]*entities.AccountBalance`. The reason being I read somewhere that generally it causes less GC thrashing to copy around actual things rather than pointers to them as they can often live on the stack then instead of the heap.